pyro/contrib/funsor/infer/traceenum_elbo.py

```diff
@@ -23,7 +23,8 @@ def terms_from_trace(tr):
         # add markov dimensions to the plate_to_step dictionary
         if node["type"] == "markov_chain":
             terms["plate_to_step"][node["name"]] = node["value"]
-        if node["type"] != "sample" or type(node["fn"]).__name__ == "_Subsample":
+        if node["type"] != "sample" or type(node["fn"]).__name__ == "_Subsample" or \
+                node["infer"].get("_do_not_score", False):
             continue
```

Comment:
After testing these changes I've found out that auxiliary sites are not added to the `measure_vars`. Something like this should fix it:

```py
        if node["type"] != "sample" or type(node["fn"]).__name__ == "_Subsample":
            continue
        if node["infer"].get("_do_not_score", False):
            if node["funsor"].get("log_measure", None) is not None:
                terms["measure_vars"] |= (frozenset(node["funsor"]["value"].inputs) | {name}) - terms["plate_vars"]
            continue
```

Reply:
Thanks, should be fixed.